### DISCUSSION TITLE
Disable commands for Browser or Codespaces environment

### DISCUSSION THREAD

**Discussion by brianignacio5** [microsoft/vscode-discussions#1581] [Extension Development QnA] [answered]
I've tried to set a when:"!isWeb" to extension contribution command to disable some commands when running on Codespaces.

The extension is installed locally (not on browser) in the codespaces case. Is there any way to disable a common when UI is Web ?

**Comment by alefragnani** [comment, marked as answer]
Hi @brianignacio5 ,

In order to detect if you are running in Codespaces, you could use this:

```typescript
export function isRunningOnCodespaces(): boolean {
    return env.remoteName?.toLocaleLowerCase() === 'codespaces';
}
```

I use this approach on my [Project Manager](https://marketplace.visualstudio.com/items?itemName=alefragnani.project-manager) extension, and it works just fine. It was discovered while developing this feature https://github.com/alefragnani/vscode-project-manager/issues/479

Hope this helps
